oidc_test.go

```diff
@@ -297,3 +299,148 @@ func TestNewProvider(t *testing.T) {
 		})
 	}
 }
+
+type userInfoTestServer struct {
```

Comment:
Instead of having userInfoTestServer and mockServer as separate structs, just have one "testServer" struct:

```go
type testServer struct {
	wellKnown   string
	contentType string
	jwks        string
	userInfo    string
}

func (t *testServer) run(t *testing.T) string {
    mux := http.NewServeMux()
    // register mux handlers
    srv := httptest.NewServer(mux)
    t.Cleanup(srv.Close)
    return srv.URL
}
```

Your test is then

```go
t.Run(test.name, func(t *testing.T) {
    serverURL := test.server.run(t)

    // logic
})
```

Reply:
In this case, it's going to start the HTTP server for every test case, which I considered costly.
I wanted to separate the startup from the single test run.

If you don't care about the start cost I can simplify it similar to what you suggested.
Should I ?

Reply:
I'm fine starting it for every test case. If it's an actual performance issue we can optimize then.

Reply:
Changed it, additionally moved some of the shared mock responses into the `run(t *testing.T)`.